apps/events/migrations/0011_modify_schedulemessage_class.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 4.2.11 on 2024-06-19 17:15
+from datetime import datetime, timedelta
+from django.db import migrations
+
+from apps.events.models import EventAction, EventActionType, ScheduledMessage
+from apps.experiments.models import ExperimentSession
+
+def is_complete(task):
+	return datetime.now() > task.expires
+
+def calculate_next_run(task, current_time, is_complete):
+	if is_complete:
+		return None
+	delta = timedelta(seconds=task.interval.every.total_seconds())
+	next_time = task.start_time
+	while next_time <= current_time:
+		next_time += delta
+	return next_time
+
+def calculate_last_run(task, current_time, is_complete):
```

Comment:
We have the last run time from the periodic task, so no need to calculate that :guitar: